File: RevShop_P2/backend/src/main/java/com/revshop/service/OrderService.java
package com.revshop.service;

import com.revshop.dto.*;
import com.revshop.model.OrderStatus;
import java.util.List;

public interface OrderService {

    // Gotam's Methods (Order Management)
    OrderResponse placeOrder(OrderRequest request);

    List<OrderResponse> getOrdersByBuyer(Long buyerId);

    List<OrderResponse> getOrdersBySeller(Long sellerId);

    OrderResponse getOrderById(Long orderId);

    OrderResponse updateOrderStatus(Long orderId, OrderStatus status);

    void cancelOrder(Long orderId);

    // Anusha's Methods (Checkout & Payment)
    OrderResponseDto checkout(CheckoutRequestDto request);

    OrderResponseDto processPayment(PaymentRequestDto request);
}

File: RevShop_P2/frontend/src/app/core/services/order.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

export interface OrderItemRequest {
  productId: number;
  quantity: number;
}

export interface OrderRequest {
  buyerId: number;
  shippingAddress: string;
  paymentMethod: string;
  items: OrderItemRequest[];
}

export interface OrderItemResponse {
  productId: number;
  productName: string;
  quantity: number;
  priceAtPurchase: number;
  subtotal: number;
}

export interface OrderResponse {
  orderId: number;
  buyerName: string;
  buyerEmail: string;
  status: string;
  totalAmount: number;
  shippingAddress: string;
  paymentMethod: string;
  orderDate: string;
  items: OrderItemResponse[];
}

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  private API = 'http://localhost:8080/api/orders';

  constructor(private http: HttpClient) {}

  placeOrder(order: OrderRequest): Observable<OrderResponse> {
    return this.http.post<OrderResponse>(this.API, order);
  }

  getOrdersByBuyer(buyerId: number): Observable<OrderResponse[]> {
    return this.http.get<OrderResponse[]>(`${this.API}/buyer/${buyerId}`);
  }

  getOrdersBySeller(sellerId: number): Observable<OrderResponse[]> {
    return this.http.get<OrderResponse[]>(`${this.API}/seller/${sellerId}`);
  }

  getOrderById(orderId: number): Observable<OrderResponse> {
    return this.http.get<OrderResponse>(`${this.API}/${orderId}`);
  }

  updateOrderStatus(orderId: number, status: string): Observable<OrderResponse> {
    return this.http.put<OrderResponse>(`${this.API}/${orderId}/status?status=${status}`, {});
  }

  cancelOrder(orderId: number): Observable<string> {
    return this.http.put(`${this.API}/${orderId}/cancel`, {}, { responseType: 'text' });
  }
}
